test/extended/image_ecosystem/mongodb_replica_petset.go

```diff
@@ -0,0 +1,130 @@
+			podNames, err := exutil.WaitForPods(
+				oc.KubeREST().Pods(oc.Namespace()),
+				exutil.ParseLabelsOrDie("name=mongodb-replicaset"),
+				exutil.CheckPodIsRunningFn,
+				3,
+				1*time.Minute,
+			)
+			o.Expect(err).NotTo(o.HaveOccurred())
+			o.Expect(podNames).Should(o.HaveLen(3))
+
+			g.By("expecting that we can insert a new record on primary node")
+			mongo := dbutil.NewMongoDB(podNames[0])
+			replicaSet := mongo.(exutil.ReplicaSet)
+			_, err = replicaSet.QueryPrimary(oc, `db.test.save({ "status" : "passed" })`)
+			o.Expect(err).ShouldNot(o.HaveOccurred())
+
+			g.By("expecting that we can read a record from all members")
+			for _, podName := range podNames {
+				// don't include _id field to output because it changes every time
+				findCmd := "rs.slaveOk(); printjson(db.test.find({}, {_id: 0}).toArray())"
+
+				fmt.Fprintf(g.GinkgoWriter, "DEBUG: reading record from pod %v\n", podName)
+
+				mongoPod := db.NewMongoDB(podName)
+				result, err := mongoPod.Query(oc, findCmd)
+				o.Expect(err).ShouldNot(o.HaveOccurred())
+				o.Expect(result).Should(o.ContainSubstring(`{ "status" : "passed" }`))
+			}
+
+			g.By("restarting replica set")
+			err = oc.AsAdmin().Run("delete").Args("pods", "--all", "-n", oc.Namespace()).Execute()
+			o.Expect(err).ShouldNot(o.HaveOccurred())
+
+			g.By("waiting for restarting of the pods")
+			podNames, err = exutil.WaitForPods(
+				oc.KubeREST().Pods(oc.Namespace()),
+				exutil.ParseLabelsOrDie("name=mongodb-replicaset"),
+				exutil.CheckPodIsRunningFn,
+				3,
+				1*time.Minute,
+			)
+			o.Expect(err).NotTo(o.HaveOccurred())
+			o.Expect(podNames).Should(o.HaveLen(3))
+
+			g.By("expecting that we can read a record from all members after its restart")
+			for _, podName := range podNames {
+				// don't include _id field to output because it changes every time
+				findCmd := "rs.slaveOk(); printjson(db.test.find({}, {_id: 0}).toArray())"
+
+				fmt.Fprintf(g.GinkgoWriter, "DEBUG: reading record from pod %v\n", podName)
+
+				mongoPod := db.NewMongoDB(podName)
+				result, err := mongoPod.Query(oc, findCmd)
+				o.Expect(err).ShouldNot(o.HaveOccurred())
+				o.Expect(result).Should(o.ContainSubstring(`{ "status" : "passed" }`))
+			}
+		})
+	})
+
+})
+
+func cleanupPetSet(oc *exutil.CLI) {
+	err := oc.AsAdmin().Run("delete").Args("petset", "--all", "-n", oc.Namespace()).Execute()
+	if err != nil {
+		fmt.Fprintf(g.GinkgoWriter, "WARNING: couldn't cleanup pet set: %v", err)
+	}
+
+	err = oc.AsAdmin().Run("delete").Args("service", "--all", "-n", oc.Namespace()).Execute()
+	if err != nil {
+		fmt.Fprintf(g.GinkgoWriter, "WARNING: couldn't cleanup service: %v", err)
+	}
+
+	err = oc.AsAdmin().Run("delete").Args("pvc", "--all", "-n", oc.Namespace()).Execute()
+	if err != nil {
+		fmt.Fprintf(g.GinkgoWriter, "WARNING: couldn't cleanup persistent volume claims: %v", err)
+	}
+
+	err = exutil.CleanupHostPathVolumes(oc.AdminKubeREST().PersistentVolumes(), oc.Namespace())
```

Comment:
if you look at the signature of the method in question, you'll see it's just using the namespace as a unique-ifying prefix for the volume names.